### PR TITLE
Fix incorrect string value when doing int_handle.toNext

### PR DESCRIPTION
**Pull request by breezewish**
Signed-off-by: Wish <breezewish@outlook.com>

### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflash/issues/6699

Problem Summary:

Consider an Int Handle in variable `foo` which is 0xCCAB. It is encoded as `{ 80 00 00 00 00 00 CC AB }` in bytes.

In #6538:

```
foo.toNext:       int_value = 0xCCAC, bytes_value = { 80 00 00 00 00 00 CC AB 00 }
foo.toPrefixNext: int_value = 0xCCAC, bytes_value = { 80 00 00 00 00 00 CC AC }
```

The implementation of `toNext()` is incorrect because it violates the rule that the int_value and bytes_value should be matching. 

Let's see what will happen when this rule is not respected:

- Consider we construct a new Int Handle in variable `bar` from int 0xCCAC. It will be built as `int_value = 0xCCAC, bytes_value = { 80 00 00 00 00 00 CC AC }`. Now `bar` and `foo`'content are not identical any more.

- The `max`, `min` and several other basic functions are implemented by comparing bytes values:

  ```c++
  inline const RowKeyValue & max(const RowKeyValue & a, const RowKeyValue & b)
  {
      return a.value->compare(*b.value) >= 0 ? a : b;
  }

  inline const RowKeyValue & min(const RowKeyValue & a, const RowKeyValue & b)
  {
      return a.value->compare(*b.value) < 0 ? a : b;
  }

  inline bool none() const
  {
      return start.value->compare(*end.value) >= 0;
  }

  inline bool intersect(const RowKeyRange & other) const
  {
      return max(other.start, start).value->compare(*min(other.end, end).value) < 0;
  }

  bool operator==(const RowKeyRange & rhs) const
  {
      return start.value->compare(*rhs.start.value) == 0 && end.value->compare(*rhs.end.value) == 0;
  }
  ```

- As a result, buggy things happen, like `foo.toNext() != bar`. Everything related with equality or orders may have wrong results.

### What is changed and how it works?

Encode Int handle using `toPrefixNext()` in `toNext()`, so that the int_value and bytes_value are still matching.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by JaySon-Huang** [comment]
can you elaborate more about why this cause data lost in #6699?

**Comment by breezewish** [comment]
> can you elaborate more about why this cause data lost in #6699?

I updated the PR description, please take a look again, thanks!

File: dbms/src/Storages/DeltaMerge/tests/gtest_key_range.cpp [hunk 1]
```diff
@@ -148,6 +148,37 @@ TEST(RowKey, ToNextKeyCommonHandle)
...
+{
+    auto v0 = RowKeyValue::fromHandle(Handle(1178400));
+    auto v0_next = v0.toNext();
+    auto v1 = RowKeyValue::fromHandle(Handle(1178401));
+
+    EXPECT_EQ(v0, min(v0, v1));
+    EXPECT_EQ(v0, min(v0, v0_next));
+
+    EXPECT_EQ(v1, max(v0, v1));
+    EXPECT_EQ(v1, max(v0, v0_next));
```
Comment (hehechen) [inline dbms/src/Storages/DeltaMerge/tests/gtest_key_range.cpp:179]: It's confused. `EXPECT_EQ(v1, v0_next)` is better ?

**Review by hehechen — APPROVED** [review]

**Review by JaySon-Huang — APPROVED** [review]
lgtm

**Pull request merged**